File: src/commands/ls.ts
import { ProjectIndex } from "../lib/index.ts";

export async function ls(indexPath: string): Promise<void> {
  const idx = await ProjectIndex.load(indexPath);
  const entries = idx.list();
  if (entries.length === 0) {
    console.error("No projects indexed. Clone a repo or run 'gx rebuild'.");
    return;
  }
  const maxName = Math.max(...entries.map((e) => e.name.length));
  for (const entry of entries) {
    console.log(`${entry.name.padEnd(maxName)}  ${entry.path}`);
  }
}

File: crates/gx/src/commands/ls.rs
//! `gx ls` — list indexed projects, name-padded. Ported from
//! `src/commands/ls.ts`.

use std::path::Path;

use crate::index_store::ProjectIndex;

pub fn ls(index_path: &Path) {
    let idx = ProjectIndex::load(index_path);
    let entries = idx.list();
    if entries.is_empty() {
        eprintln!("No projects indexed. Clone a repo or run 'gx rebuild'.");
        return;
    }
    let max_name = entries
        .iter()
        .map(|(name, _)| name.chars().count())
        .max()
        .unwrap_or(0);
    for (name, entry) in &entries {
        println!("{name:<max_name$}  {}", entry.path);
    }
}
